File: src/OrderService.Domain/Entities/Order.cs
using OrderService.Domain.Enums;

namespace OrderService.Domain.Entities;

/// <summary>
/// Represents a customer order in the mini commerce system.
/// </summary>
public class Order
{
    public Guid Id { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public string CustomerName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string ProductSku { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal TotalAmount { get; set; }
    public OrderStatus Status { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime UpdatedDate { get; set; }
}

File: frontend/src/models/order.ts
export interface Order {
  id: string;
  orderNumber: string;
  customerName: string;
  email: string;
  productSku: string;
  quantity: number;
  totalAmount: number;
  status: string;
  createdDate: string;
  updatedDate: string;
}

export interface CreateOrderRequest {
  customerName: string;
  email: string;
  productSku: string;
  quantity: number;
  totalAmount: number;
}

export interface UpdateOrderRequest {
  customerName: string;
  email: string;
  productSku: string;
  quantity: number;
  totalAmount: number;
  status: string;
}

export interface InventoryItem {
  id: string;
  productSku: string;
  productName: string;
  quantityAvailable: number;
  quantityReserved: number;
  createdDate: string;
  updatedDate: string;
}

export interface NotificationItem {
  id: string;
  orderId?: string;
  recipientEmail: string;
  subject: string;
  body: string;
  type: string;
  status: string;
  createdDate: string;
  sentDate?: string;
}
